src/Nest/Mapping/DynamicTemplate/SingleMapping.cs

```diff
@@ -61,5 +66,157 @@ public IProperty Nested<TChild>(Func<NestedPropertyDescriptor<T, TChild>, INeste
 
 		public IProperty Percolator(Func<PercolatorPropertyDescriptor<T>, IPercolatorProperty> selector) =>
 			selector?.Invoke(new PercolatorPropertyDescriptor<T>());
+
+#pragma warning disable CS3001 // Argument type is not CLS-compliant
+		public IProperty Scalar(Expression<Func<T, int>> field, Func<NumberPropertyDescriptor<T>, INumberProperty> selector) =>
+			selector?.InvokeOrDefault(new NumberPropertyDescriptor<T>().Name(field).Type(NumberType.Integer));
```

Comment:
I think all of these should be
```
selector.InvokeOrDefault(...)
```
if `selector` is `null`, we want to return the default, but the null conditional on the `.InvokeOrDefault()` will return `null` if `selector` is `null`.